src/commands/functions-discover.ts

```diff
@@ -0,0 +1,22 @@
+import { Command } from "../command";
+import { Options } from "../options";
+import { logger } from "../logger";
+import { loadCodebases } from "../deploy/functions/prepare";
+import { normalizeAndValidate } from "../functions/projectConfig";
+import { getProjectAdminSdkConfigOrCached } from "../emulator/adminSdkConfig";
+import { needProjectId } from "../projectUtils";
+
+import type { FirebaseConfig } from "../deploy/functions/args";
+
+export const command = new Command("functions:discover")
+  .description("Discover function triggers defined in the current project directory")
+  .action(async (options: Options) => {
+    const projectId = needProjectId(options);
+    const fnConfig = normalizeAndValidate(options.config.src.functions);
+    const firebaseConfig = (await getProjectAdminSdkConfigOrCached(projectId)) as FirebaseConfig;
+    const builds = await loadCodebases(fnConfig, options, firebaseConfig, {
+      firebase: firebaseConfig,
+    });
+    logger.info(JSON.stringify(builds, null, 2));
```

Comment:
Should this `JSON.stringify` be try/catch'd ?

Reply:
I feel like that edge case is rare enough for us to just rely on the CLI's global try-catch mechanism. We are going to rethrow the error anyway right?

WDYT?